curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-308-an-amazing-prime-generating-automaton.md

```diff
@@ -12,31 +12,39 @@ A program written in the programming language Fractran consists of a list of fra
 
 The internal state of the Fractran Virtual Machine is a positive integer, which is initially set to a seed value. Each iteration of a Fractran program multiplies the state integer by the first fraction in the list which will leave it an integer.
 
-For example, one of the Fractran programs that John Horton Conway wrote for prime-generation consists of the following 14 fractions:1791 , 7885 , 1951 , 2338 , 2933 , 7729 , 9523 , 7719 , 117 , 1113 , 1311 , 152 , 17 , 551 . Starting with the seed integer 2, successive iterations of the program produce the sequence: 15, 825, 725, 1925, 2275, 425, ..., 68, 4, 30, ..., 136, 8, 60, ..., 544, 32, 240, ...
+For example, one of the Fractran programs that John Horton Conway wrote for prime-generation consists of the following 14 fractions:
 
-The powers of 2 that appear in this sequence are 22, 23, 25, ... It can be shown that all the powers of 2 in this sequence have prime exponents and that all the primes appear as exponents of powers of 2, in proper order!
+$$\frac{17}{91}, \frac{78}{85}, \frac{19}{51}, \frac{23}{38}, \frac{29}{33}, \frac{77}{29}, \frac{95}{23}, \frac{77}{19}, \frac{1}{17}, \frac{11}{13}, \frac{13}{11}, \frac{15}{2}, \frac{1}{7}, \frac{55}{1}$$
 
-If someone uses the above Fractran program to solve Project Euler Problem 7 (find the 10001st prime), how many iterations would be needed until the program produces 210001st prime ?
+Starting with the seed integer 2, successive iterations of the program produce the sequence:
+
+$$15, 825, 725, 1925, 2275, 425, \ldots, 68, \mathbf{4}, 30, \ldots, 136, \mathbf{8}, 60, \ldots, 544, \mathbf{32}, 240, \ldots$$
+
+The powers of 2 that appear in this sequence are $2^2, 2^3, 2^5, \ldots$.
+
+It can be shown that all the powers of 2 in this sequence have prime exponents and that all the primes appear as exponents of powers of 2, in proper order!
+
+If someone uses the above Fractran program to solve Project Euler Problem 7 (find the ${10001}^{\text{st}}$ prime), how many iterations would be needed until the program produces $2^{10001\text{st}}$ prime?
```

Comment:
```suggestion
If someone uses the above Fractran program to solve Project Euler Problem 7 (find the ${10001}^{\text{st}}$ prime), how many iterations would be needed until the program produces $2^{10001\text{st prime}}$?
```

Not sure on this one, but that's how it looks on the [Euler site](https://projecteuler.net/problem=308)

Reply:
I remember deliberating on this.

Regarding the "prime" I didn't know what it could mean including it in exponent, so I mostly assumed that's a mistake and it should be written normally. Similarly as it is written by the `10001^st` prime just before that. Looking now again at it for a while I realize I might have been wrong.

As program generates numbers `2^2`, `2^3`, `2^5`, ...; and these exponents are 1st prime, 2nd prime, 3rd prime, etc....; So in there it actually is the 10001st (or rather 10001^st) prime. Although more could be in that case:
`$2^{10001^{\text{st}}\text{ prime}}$`
![image](https://user-images.githubusercontent.com/60067306/126365757-89c05adc-94cb-40fe-8640-dbecedee8d68.png)

Reply:
I updated the suggestion to that.